agent/agent.go

```diff
@@ -379,6 +335,93 @@ func (a *Agent) loop(ctx context.Context) error {
 	}
 }
 
+// ProcessNextResponse contains information for what should be done after processing next task
+type ProcessNextResponse struct {
+	ShouldExit       bool
+	ShouldContinue   bool
+	ShouldResetTimer bool
+	needPostGroup    bool
+	tc               *taskContext
+}
+
+// ProcessNextTask takes in the NextTaskResponse and processes it.
+func (a *Agent) ProcessNextTask(nt *apimodels.NextTaskResponse, ctx context.Context, tc *taskContext, needPostGroup bool) (ProcessNextResponse, error) {
```

Comment:
drive by review but can we make this function and ProcessNextResponse private?

Reply:
+1 to this